packages/hooks/src/mutations/add-lines.ts

```diff
@@ -41,13 +42,33 @@ export const useAddLineItemsToCartMutation = <Cart extends BarebonesCart>({
 
       surfaceMutationErrors(data, userErrors, silenceUserErrors)
 
-      if (!cartId) cartLocalStorage.set(data.id)
+      if (!cartId) {
+        cartLocalStorage.set(data.id)
+      }
 
       if (updateCartQueryDataOnSuccess) {
         optimisticCartUpdate.update(data)
       }
       return data
     },
-    options?.mutationOptions
+    {
+      ...options?.mutationOptions,
+      onError(error) {
+        if (logging?.onError) {
+          logging.onError(
+            cartId ? 'addLineItemError' : 'createCartError',
+            error as Error
+          )
+        }
+      },
+      onSuccess(data) {
+        if (logging?.onSuccess) {
+          logging.onSuccess(
+            cartId ? 'addLineItemSuccess' : 'createCartSuccess',
+            data
+          )
+        }
+      }
```

Comment:
we also need to call `options?.mutationOptions?.onError()` and `options?.mutationOptions?.onSuccess()`